.github/workflows/submit.yml

```diff
@@ -602,346 +478,70 @@ jobs:
-          tar -xf "${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin.tar.gz" -C "${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin"
-
-      - name: Find root of build JDK image dir
-        run: |
-          build_jdk_root=`find ${HOME}/jdk-linux-x64/jdk-${{ env.JDK_VERSION }}-internal+0_linux-x64_bin -name release -type f`
-          echo "build_jdk_root=`dirname ${build_jdk_root}`" >> $GITHUB_ENV
-
-      - name: Install cross-compilation host dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install g++-powerpc64le-linux-gnu gcc-powerpc64le-linux-gnu:
-
-      - name: Cache sysroot
-        id: cache-sysroot
-        uses: actions/cache@v2
-        with:
-          path: ~/sysroot-ppc64el/
-          key: sysroot-ppc64el-${{ hashFiles('jdk/.github/workflows/submit.yml') }}
-
-      - name: Install sysroot host dependencies
+      - name: Configure cross-specific flags
         run: |
-          sudo apt-get install debootstrap qemu-user-static
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
-
-      - name: Create sysroot
-        run: >
-          sudo qemu-debootstrap
-          --arch=ppc64el
-          --verbose
-          --include=fakeroot,symlinks,build-essential,libx11-dev,libxext-dev,libxrender-dev,libxrandr-dev,libxtst-dev,libxt-dev,libcups2-dev,libfontconfig1-dev,libasound2-dev,libfreetype6-dev,libpng-dev
-          --resolve-deps
-          buster
-          ~/sysroot-ppc64el
-          http://httpredir.debian.org/debian/
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
-
-      - name: Relativize symlinks in sysroot
-        run: >
-          sudo chroot ~/sysroot-ppc64el symlinks -cr .
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
-
-      - name: Fix permissions in sysroot
-        run: >
-          sudo chown ${USER} -R ~/sysroot-ppc64el
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
-
-      - name: Remove special directories in sysroot
-        run: >
-          rm -rf ~/sysroot-ppc64el/{dev,proc,run,sys}
-        if: steps.cache-sysroot.outputs.cache-hit != 'true'
+          echo "CC=${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-flavor}}-gcc-10" >> $GITHUB_ENV
+          echo "CXX=${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-flavor}}-g++-10" >> $GITHUB_ENV
+          echo "cross_flags=--openjdk-target=${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-flavor}} --with-sysroot=${HOME}/sysroot-${{ matrix.debian-arch }}/ --with-toolchain-path=${HOME}/sysroot-${{ matrix.debian-arch }}/ --with-freetype-lib=${HOME}/sysroot-${{ matrix.debian-arch }}/usr/lib/${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-flavor}}/ --with-freetype-include=${HOME}/sysroot-${{ matrix.debian-arch }}/usr/include/freetype2/ --x-libraries=${HOME}/sysroot-${{ matrix.debian-arch }}/usr/lib/${{ matrix.gnu-arch }}-linux-gnu${{ matrix.gnu-flavor}}/" >> $GITHUB_ENV
```

Comment:
No chance to break this line by option? It is kinda hard to read and thus verify. It is still okay if we cannot.

Reply:
Sure, I agree, perhaps just putting that as a separate step would increase readability.

Reply:
The update looks fine.